src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1279 @@
+            df.at[person_id, 'un_am_discharge_date'] = self.sim.date + DateOffset(weeks=3)
+            df.at[person_id, 'un_am_treatment_type'] = 'CSB++'
+            self.module.do_when_am_treatment(person_id, intervention='SFP')
+        else:
+            logger.debug(key='debug', data="PkgCode1 is not available, so can't use it.")
+
+    def did_not_run(self):
+        logger.debug("Undernutrition_Feeding_Supplementary: did not run")
+        pass
+
+
+class HSI_Wasting_OutpatientTherapeuticProgramme_SAM(HSI_Event, IndividualScopeEventMixin):
+    """
+    This is the outpatient management of SAM without any medical complications
+    """
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+        assert isinstance(module, Wasting)
+
+        # Get a blank footprint and then edit to define call on resources
+        # of this treatment event
+        the_appt_footprint = self.sim.modules["HealthSystem"].get_blank_appt_footprint()
+        the_appt_footprint['U5Malnutr'] = 1
+
+        # Define the necessary information for an HSI
+        self.TREATMENT_ID = 'Undernutrition_Feeding_Outpatient'
+        self.EXPECTED_APPT_FOOTPRINT = the_appt_footprint
+        self.ACCEPTED_FACILITY_LEVEL = '1a'
+        self.ALERT_OTHER_DISEASES = []
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+
+        # Stop the person from dying of acute malnutrition (if they were
+        # going to die)
+        if not df.at[person_id, 'is_alive']:
+            return
+
+        # Do here whatever happens to an individual during this health
+        # system interaction event
+        # ~~~~~~~~~~~~~~~~~~~~~~
+        # Make request for some consumables
+        consumables = self.sim.modules['HealthSystem'].parameters[
+            'item_and_package_code_lookups']
+
+        # individual items
+        item_code1 = pd.unique(consumables.loc[consumables['Items'] ==
+                                               'SAM theraputic foods', 'Item_Code'])[0]
+        item_code2 = pd.unique(consumables.loc[consumables['Items'] == 'SAM medicines', 'Item_Code'])[0]
+
+        # check availability of consumables
+        if self.get_consumables(item_code1) and self.get_consumables(item_code2):
+            logger.debug(key='debug', data='consumables are available.')
+            # Log that the treatment is provided:
+            df.at[person_id, 'un_am_tx_start_date'] = self.sim.date
+            df.at[person_id, 'un_am_discharge_date'] = self.sim.date + DateOffset(weeks=3)
+            df.at[person_id, 'un_am_treatment_type'] = 'standard_RUTF'
+            self.module.do_when_am_treatment(person_id, intervention='OTC')
+        else:
+            logger.debug(key='debug', data="consumables not available, so can't use it.")
+
+    def did_not_run(self):
+        logger.debug("HSI_Undernutrition_Feeding_Outpatient: did not run")
```

Comment:
Determinism test is failing because this uses the old (and now removed) way of logging. log needs arguments `key` and `data`.

Reply:
Let's just delete this obsolete line then!

Reply:
@tamuri, Why the `test_determinism` is not failing on my computer?